test/v4/controllers/SubmitFinalDeclarationControllerSpec.scala

```diff
@@ -0,0 +1,175 @@
+import api.models.audit.{AuditEvent, AuditResponse, GenericAuditDetail}
+import api.models.domain.{CalculationId, Nino, TaxYear}
+import api.models.errors.{ErrorWrapper, InternalError, NinoFormatError, RuleTaxYearNotSupportedError}
+import api.models.outcomes.ResponseWrapper
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.Result
+import v4.mocks.requestParsers.MockSubmitFinalDeclarationParser
+import v4.mocks.services._
+import v4.models.request._
+import v4.models.response.retrieveCalculation.CalculationFixture
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class SubmitFinalDeclarationControllerSpec
+    extends ControllerBaseSpec
+    with ControllerTestRunner
+    with MockEnrolmentsAuthService
+    with MockMtdIdLookupService
+    with MockSubmitFinalDeclarationService
+    with MockSubmitFinalDeclarationParser
+    with MockRetrieveCalculationService
+    with MockAuditService
+    with MockNrsProxyService
+    with MockIdGenerator
+    with CalculationFixture {
+
+  private val taxYear       = "2020-21"
+  private val calculationId = "4557ecb5-fd32-48cc-81f5-e6acd1099f3c"
+
+  trait Test extends ControllerTest with AuditEventChecking {
+
+    val controller = new SubmitFinalDeclarationController(
+      authService = mockEnrolmentsAuthService,
+      lookupService = mockMtdIdLookupService,
+      parser = mockSubmitFinalDeclarationParser,
+      service = mockSubmitFinalDeclarationService,
+      retrieveService = mockRetrieveCalculationService,
+      cc = cc,
+      nrsProxyService = mockNrsProxyService,
+      auditService = mockAuditService,
+      idGenerator = mockIdGenerator
+    )
+
+    protected def callController(): Future[Result] = controller.submitFinalDeclaration(nino, taxYear, calculationId)(fakeRequest)
+
+    override protected def event(auditResponse: AuditResponse, maybeRequestBody: Option[JsValue]): AuditEvent[GenericAuditDetail] =
+      AuditEvent(
+        "SubmitAFinalDeclaration",
+        "submit-a-final-declaration",
+        GenericAuditDetail(
+          userType = "Individual",
+          agentReferenceNumber = None,
+          params = Map("nino" -> nino, "taxYear" -> taxYear, "calculationId" -> calculationId),
+          requestBody = None,
+          `X-CorrelationId` = correlationId,
+          versionNumber = "3.0",
+          auditResponse = auditResponse
+        )
+      )
+
+  }
+
+  private val rawData                     = SubmitFinalDeclarationRawData(nino, taxYear, calculationId)
+  private val requestData                 = SubmitFinalDeclarationRequest(Nino(nino), TaxYear.fromMtd(taxYear), CalculationId(calculationId))
+  private val retrieveDetailsRequestData  = RetrieveCalculationRequest(Nino(nino), TaxYear.fromMtd(taxYear), CalculationId(calculationId))
+  private val retrieveDetailsResponseData = minimalCalculationR8bResponse
+
+  "V4-submit final declaration" should {
```

Comment:
We don't need to specify v4 as its a v4 spec. We should also update with 
```suggestion
  "SubmitFinalDeclarationController" should {
```